src/com/parse4cn1/ParseUser.java

```diff
@@ -320,11 +320,11 @@ public void logout() throws ParseException {
             ParseCommand command = new ParsePostCommand(ENDPOINT_LOGOUT);
             command.addHeader(ParseConstants.HEADER_SESSION_TOKEN, getSessionToken());
             ParseResponse response = command.perform();
+            setSessionToken(null);
```

Comment:
@ahmedengu Why would it make more sense to reset the session and current user before checking whether the action succeeded. In my opinion the original implementation makes more sense i.e. resetting only after a successful deletion